### PR TITLE
chore: add logging when emitting metrics

### PR DESCRIPTION
**Pull request by sameerzuberi**
Adding logging when the CDA emits metrics to see that metrics are being emitted, and to see which metrics are being emitted. This is to assist in debugging UAT failures.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/clientdevices/auth/api/ClientDevicesAuthServiceApi.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public boolean verifyClientDeviceIdentity(String certificatePem) {
                 isVerified = useCases.get(VerifyCertificateValidityPeriod.class).apply(certificatePem) && useCases.get(
                     VerifyIotCertificate.class).apply(certificatePem);
             }
-
+            logger.atInfo().log("Successfully verified client device identity");
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/clientdevices/auth/api/ClientDevicesAuthServiceApi.java:70]: make these new ones debug

File: src/main/java/com/aws/greengrass/clientdevices/auth/metrics/ClientDeviceAuthMetrics.java [hunk 2]
```diff
@@ -64,8 +67,8 @@ public ClientDeviceAuthMetrics(Clock clock) {
      * Emit metrics using Metric Factory.
      */
     public void emitMetrics() {
-        // TODO need to call this function on a timer
         List<Metric> retrievedMetrics = collectMetrics();
+        logger.atInfo().log("Emitting metrics: ", retrievedMetrics);
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/clientdevices/auth/metrics/ClientDeviceAuthMetrics.java:71]: debug. use `kv`

File: src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java [hunk 3]
```diff
@@ -191,6 +192,7 @@ public void subscribeToCertificateUpdates(GetCertificateRequest getCertificateRe
                 subscribeToClientCertificateUpdatesNoCSR(getCertificateRequest, keyPair.getPublic(), consumer);
             }
         } catch (NoSuchAlgorithmException e) {
+            logger.atError().cause(e).log("Failed to subscribe to certificate update");
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java:195]: this should be logged by the callers already.

File: src/main/java/com/aws/greengrass/clientdevices/auth/metrics/ClientDeviceAuthMetrics.java [hunk 4]
```diff
@@ -64,8 +67,8 @@ public ClientDeviceAuthMetrics(Clock clock) {
      * Emit metrics using Metric Factory.
      */
     public void emitMetrics() {
-        // TODO need to call this function on a timer
         List<Metric> retrievedMetrics = collectMetrics();
+        logger.atDebug().kv("Emitting metrics: ", retrievedMetrics);
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/clientdevices/auth/metrics/ClientDeviceAuthMetrics.java:71]: you need to log this.

`logger.atDebug().kv("metrics", retrivedMetrics).log("Emitting CDA metrics")`

**Review by MikeDombo — APPROVED** [review]

**Review by jcosentino11 — APPROVED** [review]

**Pull request merged**